store/searchlayer/channel_layer.go

```diff
@@ -0,0 +1,210 @@
+	member, err := c.ChannelStore.UpdateMember(cm)
+	if err == nil {
+		c.rootStore.indexUserFromID(cm.UserId)
+		channel, channelErr := c.ChannelStore.Get(member.ChannelId, true)
+		if channelErr != nil {
+			mlog.Error("Encountered error indexing user in channel", mlog.String("channel_id", member.ChannelId), mlog.Err(err))
+		} else {
+			c.rootStore.indexUserFromID(channel.CreatorId)
+		}
+	}
+	return member, err
+}
+
+func (c *SearchChannelStore) SaveMember(cm *model.ChannelMember) (*model.ChannelMember, *model.AppError) {
+	member, err := c.ChannelStore.SaveMember(cm)
+	if err == nil {
+		c.rootStore.indexUserFromID(cm.UserId)
+		channel, channelErr := c.ChannelStore.Get(member.ChannelId, true)
+		if channelErr != nil {
+			mlog.Error("Encountered error indexing user in channel", mlog.String("channel_id", member.ChannelId), mlog.Err(err))
+		} else {
+			c.rootStore.indexUserFromID(channel.CreatorId)
+		}
+	}
+	return member, err
+}
+
+func (c *SearchChannelStore) RemoveMember(channelId, userIdToRemove string) *model.AppError {
+	err := c.ChannelStore.RemoveMember(channelId, userIdToRemove)
+	if err == nil {
+		c.rootStore.indexUserFromID(userIdToRemove)
+	}
+	return err
+}
+
+func (c *SearchChannelStore) CreateDirectChannel(user *model.User, otherUser *model.User) (*model.Channel, *model.AppError) {
+	channel, err := c.ChannelStore.CreateDirectChannel(user, otherUser)
+	if err == nil {
+		c.rootStore.indexUserFromID(user.Id)
+		c.rootStore.indexUserFromID(otherUser.Id)
+	}
+	return channel, err
+}
+
+func (c *SearchChannelStore) AutocompleteInTeam(teamId string, term string, includeDeleted bool) (*model.ChannelList, *model.AppError) {
+	var channelList *model.ChannelList
+	var err *model.AppError
+
+	allFailed := true
+	for _, engine := range c.rootStore.searchEngine.GetActiveEngines() {
+		if engine.IsAutocompletionEnabled() {
+			channelList, err = c.esAutocompleteChannels(engine, teamId, term, includeDeleted)
+			if err != nil {
+				mlog.Error("Encountered error on AutocompleteChannels through SearchEngine. Falling back to default autocompletion.", mlog.Err(err))
+				continue
+			}
+			allFailed = false
+			mlog.Debug("Using the first available search engine", mlog.String("search_engine", engine.GetName()))
+			break
+		}
+	}
+
+	if allFailed {
+		mlog.Debug("Using database search because no other search engine is available")
+		channelList, err = c.ChannelStore.AutocompleteInTeam(teamId, term, includeDeleted)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return channelList, err
+}
+
+func (c *SearchChannelStore) esAutocompleteChannels(engine searchengine.SearchEngineInterface, teamId, term string, includeDeleted bool) (*model.ChannelList, *model.AppError) {
+	channelIds, err := engine.SearchChannels(teamId, term)
+	if err != nil {
+		return nil, err
+	}
+
+	channelList := model.ChannelList{}
+	if len(channelIds) > 0 {
+		channels, err := c.ChannelStore.GetChannelsByIds(channelIds)
```

Comment:
Maybe we can add the `includeDeleted` flag in the `GetChannelsByIds` and we get the results filtered

Reply:
We should move forward with this? or is not necessary

Reply:
Added the proposed change